### ISSUE TITLE
[v4.29.1] Parsing CSVs with empty fields fails with "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by aleskandro** [mikefarah/yq#1404]
**Describe the bug**
Parsing CSVs with empty fields fails with `invalid memory address or nil pointer dereference` in `v4.29.1`, not in `v4.28.2`.

Version of yq: 4.29.1
Operating system: linux/arm64
Installed via: GitHub release binary 

**Input CSV**
data1.yml:
```csv
col1,col2,col3,col4
field1,field2,,field4
```

data2.csv
```csv
col1,col2,col3,col4
field1,field2,x,field4
```

If the third field above is filled with anything, it works.

**Command**
The command you ran:
```
./yq4.29.1 -p csv data1.csv
```

**Actual behavior**
``` shell
$ ./yq4.29.1 -p csv data1.csv
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3282bc]

goroutine 1 [running]:
github.com/mikefarah/yq/v4/pkg/yqlib.parseSnippet({0x400023c408, 0x0})
	/home/runner/work/yq/yq/pkg/yqlib/lib.go:258 +0xdc
github.com/mikefarah/yq/v4/pkg/yqlib.(*csvObjectDecoder).convertToYamlNode(0x408140?, {0x400023c408, 0x0})
	/home/runner/work/yq/yq/pkg/yqlib/decoder_csv_object.go:32 +0x2c
github.com/mikefarah/yq/v4/pkg/yqlib.(*csvObjectDecoder).createObject(0x4000214690?, {0x40002216c0, 0x4, 0x15?}, {0x4000221700, 0x4, 0x1?})
	/home/runner/work/yq/yq/pkg/yqlib/decoder_csv_object.go:46 +0x160
github.com/mikefarah/yq/v4/pkg/yqlib.(*csvObjectDecoder).Decode(0x400026e300)
	/home/runner/work/yq/yq/pkg/yqlib/decoder_csv_object.go:67 +0x250
github.com/mikefarah/yq/v4/pkg/yqlib.(*streamEvaluator).Evaluate(0x4000369c70, {0xffffec7ff67e, 0x5}, {0x4fcaf8?, 0x400021a5a0?}, 0x2?, {0x4fed18, 0x400021a480}, {0x4fdee8, 0x400026e300})
	/home/runner/work/yq/yq/pkg/yqlib/stream_evaluator.go:93 +0x9c
github.com/mikefarah/yq/v4/pkg/yqlib.(*streamEvaluator).EvaluateFiles(0x42f900?, {0x0, 0x0}, {0x400023ea80, 0x1, 0x4000369b28?}, {0x4fed18, 0x400021a480}, {0x4fdee8, 0x400026e300})
	/home/runner/work/yq/yq/pkg/yqlib/stream_evaluator.go:65 +0x144
github.com/mikefarah/yq/v4/cmd.evaluateSequence(0x4000244300?, {0x400023ea80, 0x1, 0x3})
	/home/runner/work/yq/yq/cmd/evalute_sequence_command.go:132 +0x6e4
github.com/spf13/cobra.(*Command).execute(0x4000244300, {0x400023ea20, 0x3, 0x3})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x5e0
github.com/spf13/cobra.(*Command).ExecuteC(0x4000244000)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x368
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
main.main()
	/home/runner/work/yq/yq/yq.go:22 +0x1f0
```

**Expected behavior**

```shell
$ ./yq4.28.2 -p csv data1.csv
- col1: field1
  col2: field2
  col3: ""
  col4: field4


```

**Comment by mikefarah** [comment]
Argh! Yep that's def a bug, will try to get it fixed soon

**Comment by mikefarah** [comment]
Fixed in 4.29.2
